src/app/features/registration/registration.component.html

```diff
@@ -1 +1,39 @@
-<p>registration works!</p>
+<navbar></navbar>
+
+<form [formGroup]="form" class="form-content" (ngSubmit)="form.valid ? registration() : null">
+  <div class="div-column">
+    <h3>Registration</h3>
+    <div class="input-group">
+      <label><b>Name</b></label>
+      <input type="text" class="input-value" required minlength="6" formControlName="name"
+         [class.danger]="(hasError || hasError1) || ((name.errors != null && (name.dirty || name.touched)))" />
+      <span class="danger" *ngIf="hasError || (name.errors?.required  && (name.dirty || name.touched))">Name is required</span>
+      <span class="danger" *ngIf="hasError1 || (name.errors?.minlength && (name.touched))"> Name is less than 6 characters </span>
+    </div>
+
+    <div class="input-group">
+      <label><b>Email</b></label>
+      <input type="text" class="input-value" required formControlName="email"
+             [class.danger]="hasError2 || (email.errors != null && (email.dirty || email.touched))" />
+      <span class="danger" *ngIf="hasError2 || (email.errors?.required  && (email.dirty || email.touched))">Name is required</span>
+    </div>
+
+    <div class="input-group">
+      <label><b>Password</b></label>
+      <input class="input-value" required type="password" formControlName="password"
+             [class.danger]="hasError3 || (password.errors != null && (password.dirty || password.touched))" />
+       <span  class="danger" *ngIf="hasError3 || (password.errors && (password.dirty || password.touched))">Password is required</span>
+    </div>
+
+     <app-button [btnText]="btnText" (click)="
+      hasError = name.errors != null && name.errors['required'] != null;
+      hasError1 = name.errors != null && name.errors['minlength'] != null;
+      hasError2 = email.errors != null;
+      hasError3 = password.errors !=null;
```

Comment:
all conditions should be moved to .ts file, template file should not contain any logic